.vscode/tasks.json

```diff
@@ -56,6 +56,20 @@
       },
       "problemMatcher": []
     },
+    {
+      "label": "Pylint (enforce_type_hint)",
+      "type": "shell",
+      "command": "pylint --disable=all --enable=hass_enforce_type_hints --ignore-missing-annotations=n homeassistant/components/${input:integrationName}",
```

Comment:
Why not just run the full Pylint against an integration?

As in, is this is merged, we will end up having two options to run Pylint:

- Pylint, all rules, against whole Home Assistant
- Pylint, just the enforce typehint rules, against a single integration.

That feels odd and inconsistent.

Reply:
Some type hint checks are disabled by default in pylint, as we didn't want to check untyped methods and put newcomers off. So in this task we need to add `--ignore-missing-annotations=n`

Regarding the integration, it is optional and leaving blank will run against all integrations.

This step `--disable=all --enable=hass_enforce_type_hints` just makes it quicker to run but we could remove it.

Reply:
If we wanted to keep just one task, I guess we could add `--ignore-missing-annotations=n` to the standard pylint task?

Or maybe even invert the logic so that it default to False in the plugin:
- we adjust `pre-commit` is use `ignore-missing-annotations=y`
- we adjust `CI` is use `ignore-missing-annotations=y`
- vscode uses default settings and shows a warning in Intellisense
- vscode task uses default settings and shows the warnings

Reply:
I like that last one better to be honest.
It would still pass in CI, but we do encourage it when developing.